Add multiturn documentation (#190)
pylhc/pylhc.github.io: docs/guis/multiturn/gui.md

# The Multiturn GUI

The Multiturn GUI provides functionality to set up and perform beam excitation with the AC Dipole or the ADT.
Excitations feature automatic saving of turn-by-turn BPM data.
This section will guide you through the GUI's layout and functionality.

The GUI is a Java application and is typically run from the `CCM`:

=== "From the CCM"
    Have a [working `CCM`][gui_basics] running as `lhcop`, then navigate to `LHC Control` -> `LHC Beam Measurements` -> `Multiturn`.

After opening, the GUI should look like this:

<figure>
  <center>
  <img src="../../assets/images/multiturn_gui/default_view.png" width="85%" alt="Multiturn GUI landing page" />
  <figcaption> Multiturn GUI Landing Page </figcaption>
  </center>
</figure>

After opening, select at the top of the GUI either the `Acquisition BEAM1` or `Acquisition BEAM2` tab, depending on the beam you plan on measuring.

!!! warning
    It is recommended to not kick both beams from the same GUI, as it can lead to crashes and unexpected behavior.
    Open a separate GUI for each beam.

> 

Reviewer: add

```
!!! tip "Closing Tabs"
   To avoid kicking the wrong beam accidentally, one can drag and drop the tab for the unused beam out of the GUI. This is especially nice when kicking Beam 1, as one often automatically lands on the Acquisition Beam 2 tab when closing the kick measurement results tabs.
 ```

Reviewer 2: Never knew this, TIL!

# The Multiturn GUI

The Multiturn GUI provides functionality to perform the basic BPM measurements in the LHC, while kicking the AC-dipole or the ADT. It allows one to measure the turn-by-turn response of the beam, during excitation and save the results for further analysis.

> This section will guide you through the GUI's layout and functionality. We also provide a procedure page for [performing Multiturn measurements in the CCC][multiturn_measurements].

Reviewer 2: Not sure what the link is supposed to send to, this is currently broken.

The GUI is a Java application and is typically run from the `CCM`:
=== "From the CCM"
    Have a [working `CCM`][gui_basics] running as `lhcop`, then navigate `LHC Control` -> `LHC Beam Measurements` -> `Multiturn`.

After opening, the GUI should look like this:
<figure>
  <center>
  <img src="../../assets/images/multiturn_gui/default_view.png" width="85%" alt="Multiturn GUI landing page" />
  <figcaption> Multiturn GUI Landing Page </figcaption>
  </center>
</figure>

## Steps to Setup the GUI

> 1. On first opening, the first step is to select the beam that you would like to work with. Select the tab at the top of the GUI named `Acquisition BEAM1` or `Acquisition BEAM2`, depending on the beam you are measuring. Do not kick both beams in the same GUI session, as this can lead to crashes and unexpected behavior. Open a separate GUI for each beam.

Reviewer 2: Would prefer simple bullet list here. Unfortunately mkdocs (or just markdown?) is a bit weird and despite your correct numbering the entries reset to `1` after each picture when rendering.

> 1. On first opening, the first step is to select the beam that you would like to work with. Select the tab at the top of the GUI named `Acquisition BEAM1` or `Acquisition BEAM2`, depending on the beam you are measuring. Do not kick both beams in the same GUI session, as this can lead to crashes and unexpected behavior. Open a separate GUI for each beam.

Reviewer 2: Conciseness as well.

```suggestion
1. After opening, select at the top of the GUI either the `Acquisition BEAM1` or `Acquisition BEAM2` tab, depending on the beam you plan on measuring. It is recommended to not kick both beams from the same GUI, as it can lead to crashes and unexpected behavior. Open a separate GUI for each beam.
```

> 2. Next, it is necessary to select an active kick group. This is done by clicking the `Select Active group` button in the top left corner of the GUI. This will open the following dialog:

Reviewer 2: ```suggestion
2. Next, select or create a kick group. This is done by clicking the `Select Active group` button in the top left corner of the GUI, which will open the following dialog:
```

<figure>
    <center>
    <img src="../../assets/images/multiturn_gui/select_kick_group.png" width="85%" alt="Select Active Group Dialog" />
    <figcaption> Select Active Group Dialog </figcaption>
    </center>
</figure>

> 3. In the dialog, typically you will want to create a new kick group. To do this, click the `Create New Group` button at the bottom in the centre. This will open the following dialog:

Reviewer 2: ```suggestion
3. Typically one wants to create a new kick group. To do so, click the `Create New Group` button at the bottom in the centre. This will open the following dialog, with a default naming scheme:
```

<figure>
    <center>
    <img src="../../assets/images/multiturn_gui/create_kick_group.png" width="85%" alt="Create New Group Dialog" />
    <figcaption> Create New Group Dialog </figcaption>
    </center>
</figure>

> 4. Change the text under `Group Name`. This name should describe the measurements that you will take in this group, and should always have the data and the beam number to start with, e.g. `YYYY-MM-DD_BEAM1_Measurement_description`. Make sure you press enter after typing the name, and then click the `Create` button. 

Reviewer 2: ```suggestion
4. Adapt the text entry under `Group Name` to reflect the measurements to be done in this group. A good naming practice is to lead with the date and beam number as suggested, e.g. `YYYY-MM-DD_BEAM1_Measurement_description`. Make sure to press ++"Enter"++ after typing the name. Optionally add a description in the field below, and click the ++"Create"++ button.
```

> 5. Once the group is created, it will appear in the list of available groups right at the bottom of the list. Then select the group by clicking on it and then clicking the `Activate Selected`. This should then create a new entry in the `LHC-OMC` logbook. <!-- Add link here? -->

Reviewer 2: Don't think a link is necessary, we have plenty referring to the logbook on this site.

```suggestion
5. Once created the new group will appear at the bottom of the list of available groups. Select it and click the `Activate Selected`. This should then create a new entry in the `LHC-OMC` logbook.
```

> 6. Next, it is necessary to set up the tunes. This is possible by clicking the buttons `Acquire QH` and `Acquire QV`. Then, the values can be manually refined if necessary.

Reviewer 2: ```suggestion
6. Next, set up the tunes in the `Tunes set-up` section on the left side of the GUI. Clicking the `Acquire QH` and `Acquire QV` buttons will update the value to the current measured one. These values can be manually refined if necessary.
```

> 

Reviewer 2: Probably here we also mention the bunch selection under `Bunches`.

> 7. Finally, before moving to the measurement panels, change the input field turns to the number of turns you want to measure. For AC-dipole measurements, this is typically **6,600 turns**, while for ADT AC-dipole measurements, this is typically **40,000 turns**. Do not set these values higher than these for the respective measurements, as this can lead to the AC-dipole being damaged or the BPM buffers overflowing causing data to be lost/overwritten.

Reviewer 2: ```suggestion
7. Finally, set the number of turns to measure in the `Turns` field below. For AC Dipole measurements, this is typically **6,600 turns**, while for ADT AC-dipole measurements it is typically **40,000 turns**. Do not set these values higher than these for the respective measurements, as this can lead to the AC-dipole being damaged or the BPM buffers overflowing causing data to be lost or overwritten.
```

> 

Reviewer 2: I think a section could be good here to mention the various information and color coding we see in `Flag Status` and `Measurement Environment`.

Author: Have a look at what I did, wasn't sure about the chroma state and landau damping

The following pages are available:

- [AC-Dipole Measurements](acdipole.md) for how to excite the beam with the AC-dipole.
- [ADT AC-Dipole Measurements](adt.md) for how to excite the beam with the ADT.

> - [AC-Dipole Scheduler](scheduler.md) for how to schedule and run AC-Dipole measurements with a set of predefined kick amplitudes.

Reviewer 2: I would add at the end of the page, after a blank line:

```markdown
*[AC Dipole]: Alternating Current Dipole
*[ADT]: LHC Transverse Damper
*[BPM]: Beam Position Monitor
```

Reviewer 2: You also need (so the link for [working CCM] above works):

```markdown
[gui_basics]: ../about.md#running-in-the-ccc-in-2025
```
